blocks/inner-blocks/index.js

```diff
@@ -0,0 +1,18 @@
+/**
+ * WordPress dependencies
+ */
+import { withContext } from '@wordpress/components';
+
+function InnerBlocks( { BlockList, layouts } ) {
+	return <BlockList layouts={ layouts } />;
+}
+
+InnerBlocks = withContext( 'BlockList' )()( InnerBlocks );
+
+InnerBlocks.Content = ( { BlockContent } ) => {
+	return <BlockContent />;
```

Comment:
Why did you choose to rely on the context instead of making it explicit, passing the `innerBlocks` and using them here? Since I assume both `edit` and `save` receive those? I have a personal preference for explicitness but I'm probably missing some context here.

Reply:
I see two reasons:

- Because if we pass the inner blocks explicitly, it might give the illusion that it's a controlled input, meaning we'd similarly want to handle `onChange`, which we (currently) have no capability of doing.
   - That said, `<BlockList onChange={ ... } />` doesn't sound like an awful idea in general, and we could "fake" it with a noop and context until it's implementable
- The more pertinent point (one which I've not argued strongly for, but I know @mtias is advocating) is the desire for an isolation between a parent and its nested blocks. As implemented here a block has no need or ability to introspect or manage its nested content.
- Minor point: It would require additional props `innerBlocks` and `setInnerBlocks`.
   - These existed in earlier commits of the branch.

Reply:
Fine enough, but we can't still make it explicit without the need to pass the inner blocks (which would assume an ònChange`), we could just pass the block `uid` which is also available as a prop I think.

Reply:
Anyway, it's not that important, just wanted to get rid of this Provider because it's not easy to follow where and how it's applied. I guess it's a common pitfall for all context based components.